### PR TITLE
feat: Add support Nightly Connect

### PR DESCRIPTION
**Pull request by Pawel-Szydlo**

### CODE REVIEW

**Comment by wottpal** [comment]
Thanks for rebasing @Pawel-Szydlo! Can you remove the version modifications in the `package.json` as well? Current one from main is `"version": "0.2.0"`. – This is done automatically by CI/Workflow via [changeset](https://github.com/changesets/changesets). 

Try to test it today!

File: src/helpers/getNightlyAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+let _adapter: NightlyConnectAdapter | undefined
+export const getNightlyConnectAdapter = async (persisted = true) => {
+  if (_adapter) return _adapter
+  _adapter = await NightlyConnectAdapter.build(
+    {
+      appMetadata: {
+        name: 'Use-inkathon selector',
+        description: 'Use-inkathon selector',
+        icon: 'https://inkathon.xyz/favicons/apple-touch-icon.png',
```
Comment (wottpal) [inline src/helpers/getNightlyAdapter.ts:11]: This needs to be dynamic. Can you add `appName: string, appIcon?: string` to the function parameters in `getNightlyConnectAdapter`.

File: assets/wallet-logos/nightly@128w.png
Comment (wottpal) [inline assets/wallet-logos/nightly@128w.png:1]: These new icons do not match the padding style of the existing ones, can you apply `20px` padding to the `512w` ones and `5px` to the `128w` ones, respectively? Same comment applies to all 4 icons.

File: src/wallets.ts
Comment (wottpal) [inline src/wallets.ts:1]: I would suggest we remove `nightly` (icons as well) completely from the wallet list and keep `nightlyConnect` for both (extension & mobile). Wdyt?

File: src/wallets.ts [hunk 2]
```diff
@@ -143,10 +168,35 @@ export const isWalletInstalled = (wallet: SubstrateWallet) => {
  */
 export const enableWallet = async (wallet: SubstrateWallet, appName: string) => {
   if (!isWalletInstalled(wallet)) return undefined
-
   try {
     if (typeof window === 'undefined') return undefined
     const injectedWindow = window as Window & InjectedWindow
+    if (wallet.id === nightlyConnect.id) {
```
Comment (wottpal) [inline src/wallets.ts:174]: Minor: Add comment above about special handling for nightly.

File: src/provider.tsx [hunk 3]
```diff
@@ -226,6 +214,8 @@ export const UseInkathonProvider: FC<UseInkathonProviderProps> = ({
     unsubscribeAccounts?.()
     setUnsubscribeAccounts(undefined)
     setActiveExtension(undefined)
+    const adapter = await getNightlyConnectAdapter()
+    await adapter?.disconnect()
```
Comment (wottpal) [inline src/provider.tsx:218]: I would suggest we wrap this in an if-block, whether `activeExtension.name === nightlyConnect.id` (untested).

**Review by wottpal — CHANGES_REQUESTED** [review]
Hey @Pawel-Szydlo, great progress! I'm pretty sure we can merge this very soon. 🚀

Added a few last comments.

One additional thing I would like to achieve is make the dependency `@nightlylabs/wallet-selector-polkadot` optional, and leave the choice to the dApp developer whether to use it or not. Also, this library is used by more than just Aleph Zero users which could lead to confusions for them. How could we achieve this? 

- [ ] In `provider.tsx`'s `disconnect` function, check whether the `activeExtension` is Nightly Connect and if yes, import the dependency dynamically & disconnect. – Could that work?
- [ ] Same in `wallets.ts` in `enableWallet`.
- [ ] Make the `peerDependency` optional via https://stackoverflow.com/a/66228639

File: src/wallets.ts
Comment (Pawel-Szydlo) [reply src/wallets.ts:1]: ok, I think that's a good idea

Comment (Pawel-Szydlo) [reply src/wallets.ts:1]: However, on further reflection I would like nightlyWallet to remain, as nightlyConnect will be an optional library.

File: src/provider.tsx (hunk 3 above)
Comment (Pawel-Szydlo) [reply src/provider.tsx:218]: unfortunately activeExtension is undefined in the disconnect function

**Comment by Pawel-Szydlo** [comment]
> Dodatkową rzeczą, którą chciałbym osiągnąć, jest uczynienie zależności `@nightlylabs/wallet-selector-polkadot`opcjonalną i pozostawienie programiście dApp wyboru, czy z niej skorzystać, czy nie. Ponadto z tej biblioteki korzystają nie tylko użytkownicy Aleph Zero, co może powodować dla nich zamieszanie. Jak moglibyśmy to osiągnąć?

Practically, this is already a universal selector because it detects wallets that are injected into web3.
Soon it will be more universal (I think about 2-3 weeks)

**Pull request merged**